pkg/scheduler/objects/application.go

```diff
@@ -973,7 +974,10 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, allowPreemption
 			// the iterator might not have the node we need as it could be reserved, or we have not added it yet
 			node := getNodeFn(requiredNode)
 			if node == nil {
-				rateLimitedLog.Warn("required node is not found (could be transient)",
+				if appLog == nil {
```

Comment:
Just replace this with getAppLog().Warn(...). No need to check for nil.

Reply:
Apologies, I was still working on it. Maybe I should switch the PR to draft, and mark it ready for review when it's done.

Reply:
Yes, that's generally the preferred approach. Also, once the PR is ready for review, it's best not to force push - we squash all commits when merging anyway, and force-pushing makes incremental reviewing harder.

Reply:
Sounds good.

Reply:
Done.